src/main/java/vi/viredis/pool/RedisPool.java

```diff
@@ -0,0 +1,127 @@
+package vi.viredis.pool;
+
+import vi.viredis.client.ViRedis;
+
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedTransferQueue;
+import java.util.concurrent.TimeUnit;
+
+/*
+ *  implements the method of common pool interface
+ */
+public class RedisPool implements Pool {
+    public boolean isShutdown;
+    private int size;
+    private String host;
+    private int port;
+    private BlockingQueue<ViRedis> queue;
+
+    /*
+     *  initializes the pool with given parameters
+     *
+     *  @param size
+     *         size of the pool
+     *  @param host
+     *         redis server host
+     *  @param port
+     *         redis server port
+     *  @throws RedisException
+     *          if there is connectivity issue with the redis server
+     */
+    public RedisPool(int size, String host, int port) throws RedisException {
+        this.size = size;
+        this.host = host;
+        this.port = port;
+        init();
+    }
+
+    /*
+     * initiate the pool with fix size
+     */
+    private void init() throws RedisException {
+        queue = new LinkedTransferQueue();
+        for (int i = 0; i < size; i++) {
+            ViRedis viRedis = createRedisConnection(host, port);
+            if(viRedis != null) queue.add(viRedis);
+        }
+    }
+
+    /*
+     * created the redis connection
+     * @param host
+     *        redis server host name
+     * @param port
+     *        redis server port
+     * @throws RedisException
+     */
+    private ViRedis createRedisConnection(String host, int port) throws RedisException {
+        try {
+            return  new ViRedis(host, port);
+        } catch (IOException e) {
+            throw new RedisException("unable to connect to the redis server");
+        }
+    }
+
+    /*
+     * returns the pooled redis connection object
+     * @throws RedisException
+     */
+    @Override
+    public ViRedis get() throws RedisException {
```

Comment:
Rename this method as commented above. Use proper formatting in this method. Rename variable 'redisObject' to redisConnection. Instead of illegalStateException throw RedisException.